lib/puppet/parser/functions/create_hocon_settings.rb

```diff
@@ -0,0 +1,90 @@
+#
+# create.conf_settings.rb
+#
+
+module Puppet::Parser::Functions
+  newfunction(:create_hocon_settings, :type => :statement, :doc => <<-EOS
+Uses create_resources to create a set of hocon_setting resources from a hash:
+
+    $settings = {
+      foo.bar => {
+        baz => 'foobarbaz'
+      },
+      one.two => {
+        three1 => {
+          value => 'one-two-three.1',
+          type  => 'string',
+        },
+        three2 => {
+          ensure => absent
+        }
+      }
+    }
+    $defaults = {
+      path => '/tmp/foo.conf'
+    }
+    create_hocon_settings($settings,$defaults)
+
+
+Will create the following resources
+
+    hocon_setting{'/tmp/foo.conf foo.bar.baz':
+      ensure  => present,
+      setting => 'foo.bar.baz',
+      value   => 'foobarbaz',
+      path    => '/tmp/foo.conf',
+    }
+    hocon_setting{'/tmp/foo.conf one.two.three1':
+      ensure  => present,
+      setting => 'one.two.three1',
+      value   => 'one-two-three.1',
+      path    => '/tmp/foo.conf',
+    }
+    hocon_setting{'/tmp/foo.conf one.two.three2':
+      ensure  => absent,
+      setting => 'one.two.three2',
+      path    => '/tmp/foo.conf',
+    }
+
+  EOS
+  ) do |arguments|
+
+    raise(Puppet::ParseError, "create_hocon_settings(): Wrong number of arguments " +
+        "given (#{arguments.size} for 1 or 2)") unless arguments.size.between?(1,2)
+
+    settings = arguments[0]
+    defaults = arguments[1] || {}
+
+    if [settings,defaults].any?{|i| !i.is_a?(Hash) }
+      raise(Puppet::ParseError,
+            'create_hocon_settings(): Requires all arguments to be a Hash')
```

Comment:
What I meant is that the user gave several arguments, one of them was not a hash - the error does not indicate which one, and what it got instead. The user has to investigate all of them.